.github/workflows/ci.yml

```diff
@@ -80,37 +100,48 @@ jobs:
         uses: actions/setup-python@v2
         with:
           python-version: 3.8
-      - name: Install MSRV Rust
+      - name: Install Rust
         uses: actions-rs/toolchain@v1
         with:
-          profile: minimal
           toolchain: 1.48.0
+          profile: minimal
           default: true
-      - name: Install maturin, poetry, and toml
-        run: pip install maturin poetry toml
-      - name: Create an isolated example directory
-        run: cp -r examples/simple-extension/ ../simple-extension-msrv
-      - name: Edit Cargo.toml and change the path of rust-numpy
+      - uses: Swatinem/rust-cache@v1
+        with:
+          working-directory: examples/simple-extension
+        continue-on-error: true
+      - name: Install toml
+        run: pip install toml
+      - name: Edit Cargo.toml and detach from workspace
         run: |
           import toml
           cargo_toml = toml.load("Cargo.toml")
-          cargo_toml["dependencies"]["numpy"]["path"] = "../rust-numpy"
           cargo_toml["dependencies"]["ndarray"] = "0.13.1"
           cargo_toml["dependencies"]["num-complex"] = "0.2.4"
+          cargo_toml["workspace"] = {}
           with open("Cargo.toml", "w") as f:
             toml.dump(cargo_toml, f)
-        working-directory: ../simple-extension-msrv
+        working-directory: examples/simple-extension
         shell: python
-      - name: Use ndarray 0.13.1
+      - name: Generate lockfile
+        run: cargo generate-lockfile
+        working-directory: examples/simple-extension
+      - name: Unify dependencies on ndarray to 0.13.1
         run: |
-          cargo generate-lockfile
-          cargo update -p $(cargo pkgid -p ndarray 2>&1 >/dev/null | grep 0.15 | sed -e 's/^[ \t]*//') --precise 0.13.1
-        working-directory: ../simple-extension-msrv
-      - name: Test Example
+          import toml
+          import subprocess
+          cargo_lock = toml.load("Cargo.lock")
+          for pkg in cargo_lock["package"]:
+            if pkg["name"] == "ndarray" and pkg["version"] != "0.13.1":
+              pkg_id = pkg["name"] + ":" + pkg["version"]
+              subprocess.run(["cargo", "update", "--package", pkg_id, "--precise", "0.13.1"], check=True)
```

Comment:
I personally like the previous shell hack, but this version looks also OK

Reply:
I have to admit that I not very fond of shell oneliners. Especially in this case where a human-readable error message has to be ignored and the current ndarray version 0.15 was hard-coded. And since we have Python at hand in any case, it felt nicer to uniformly handle all the Cargo manipulations using the toml package.